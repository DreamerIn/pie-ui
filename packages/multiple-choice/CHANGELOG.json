[
  {
    "type": "feat",
    "scope": null,
    "subject": "upgrade to latest babel",
    "merge": null,
    "header": "feat: upgrade to latest babel",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "ff89bbe6795bd2498e2c87ab3a936cb6efa78a6b",
    "gitTags": " (feature/babel-7)",
    "committerDate": "2019-02-17 22:57:13 +0000",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.7.0"
  },
  {
    "type": "feat",
    "scope": "multiple-choice",
    "subject": "fixed some issues with the design for the multiple-choice",
    "merge": null,
    "header": "feat(multiple-choice): fixed some issues with the design for the multiple-choice",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "369c98e0299ab3919a4ecf43f4ebb444cb029acd",
    "gitTags": "",
    "committerDate": "2019-02-13 20:31:28 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.6.0"
  },
  {
    "type": "chore",
    "scope": null,
    "subject": "test build",
    "merge": null,
    "header": "chore: test build",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "7d622a3793372041436c905ff64fcf8745268f55",
    "gitTags": "",
    "committerDate": "2019-02-11 16:04:16 +0000",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.2"
  },
  {
    "type": "chore",
    "scope": null,
    "subject": "test build",
    "merge": null,
    "header": "chore: test build",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "001881048a6f78b314698fe477d0de3598a6c289",
    "gitTags": "",
    "committerDate": "2019-02-11 13:44:25 +0000",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.2"
  },
  {
    "type": "chore",
    "scope": null,
    "subject": "rename log",
    "merge": null,
    "header": "chore: rename log",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "2eafb92d318b5a89bd859cbad3f2a40c32164a3b",
    "gitTags": "",
    "committerDate": "2019-02-11 12:29:23 +0000",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.2"
  },
  {
    "type": "fix",
    "scope": null,
    "subject": "tag version of math inline",
    "merge": null,
    "header": "fix: tag version of math inline",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "8ef42e50056f72dae3b91bca2a10516bc8e68855",
    "gitTags": "",
    "committerDate": "2019-02-08 10:23:00 +0000",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.1"
  },
  {
    "type": "feat",
    "scope": null,
    "subject": "material-ui version bump",
    "merge": null,
    "header": "feat: material-ui version bump",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "af326373e937eb73e91d56d2b892413c36b4aadb",
    "gitTags": "",
    "committerDate": "2019-02-07 19:20:25 +0000",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.0"
  },
  {
    "type": "fix",
    "scope": "multiple-choice",
    "subject": "fixed tests",
    "merge": null,
    "header": "fix(multiple-choice): fixed tests",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "13823ebcb63e16ee8d573fe7adbee03144b14b27",
    "gitTags": "",
    "committerDate": "2019-01-11 20:00:58 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.0"
  },
  {
    "type": "fix",
    "scope": "multiple-choice",
    "subject": "fixed tests",
    "merge": null,
    "header": "fix(multiple-choice): fixed tests",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "1cd4e71bb90a6b6caf3802da8365475d05e15026",
    "gitTags": "",
    "committerDate": "2019-01-11 19:59:19 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.0"
  },
  {
    "type": "feat",
    "scope": "multiple-choice",
    "subject": "added none option for the choice label",
    "merge": null,
    "header": "feat(multiple-choice): added none option for the choice label",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "9dbf5dec0991b3d4d55a0bebff3903997ac19755",
    "gitTags": "",
    "committerDate": "2019-01-11 19:44:04 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.0"
  },
  {
    "type": "feat",
    "scope": "multiple-choice",
    "subject": "updated the config-ui package version",
    "merge": null,
    "header": "feat(multiple-choice): updated the config-ui package version",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "d0f64cd89dd8ebc1d4d08962e5b562a80222e08d",
    "gitTags": " (origin/feat/multiple-choice-with-layout)",
    "committerDate": "2019-01-08 11:45:27 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.0"
  },
  {
    "type": "feat",
    "scope": "multiple-choice",
    "subject": "wrapped the element with the preview layout",
    "merge": null,
    "header": "feat(multiple-choice): wrapped the element with the preview layout",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "d6ad6da507f875d13904181c1cb8215d4aba0121",
    "gitTags": "",
    "committerDate": "2019-01-08 11:15:18 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.5.0"
  },
  {
    "type": "fix",
    "scope": "multiple-choice",
    "subject": "failing snapshot",
    "merge": null,
    "header": "fix(multiple-choice): failing snapshot",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "2a7abed04cdd127a956401bef8134c967a5e82cf",
    "gitTags": " (origin/issue4/multiple-choice-partial-not-clear)",
    "committerDate": "2018-12-04 14:03:56 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.4.0"
  },
  {
    "type": "fix",
    "scope": "multiple-choice",
    "subject": "changed jest regex, added snapshot tests",
    "merge": null,
    "header": "fix(multiple-choice): changed jest regex, added snapshot tests",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "69df63ba5a52b7a41b2bb33a409c82a132f6a17a",
    "gitTags": "",
    "committerDate": "2018-12-04 13:59:38 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.4.0"
  },
  {
    "type": "feat",
    "scope": "multiple-choice",
    "subject": "changed the code in order to be obvious when a user answers partially correct",
    "merge": null,
    "header": "feat(multiple-choice): changed the code in order to be obvious when a user answers partially correct",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "5fc2ccd09503cd81d3ac7e142bca33d7eca78b0f",
    "gitTags": "",
    "committerDate": "2018-11-30 10:44:10 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.4.0"
  },
  {
    "type": "fix",
    "scope": "added-dependencies-where-needed",
    "subject": "fixed missing dependencies",
    "merge": null,
    "header": "fix(added-dependencies-where-needed): fixed missing dependencies",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "b60f1729ba7cdfdda1a5f6f8e07ca4821bb62f07",
    "gitTags": "",
    "committerDate": "2018-11-09 16:43:33 +0200",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.3.2"
  },
  {
    "type": "fix",
    "scope": null,
    "subject": "render all math as single dollar is now disabled",
    "merge": null,
    "header": "fix: render all math as single dollar is now disabled",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "5caea57e237d21f7d89f306c56ca44663e41d46e",
    "gitTags": "",
    "committerDate": "2018-09-13 21:56:23 +0100",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.3.1"
  },
  {
    "type": "feat",
    "scope": null,
    "subject": "upgrade to @pie-lib/math-rendering@^1.0.0",
    "merge": null,
    "header": "feat: upgrade to @pie-lib/math-rendering@^1.0.0",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "3e86ddd723236b82e8118d171bbbd18b402e9dcf",
    "gitTags": "",
    "committerDate": "2018-09-13 19:33:21 +0100",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.3.0"
  },
  {
    "type": "fix",
    "scope": null,
    "subject": "use @pie-lib/math-rendering@0.5.0",
    "merge": null,
    "header": "fix: use @pie-lib/math-rendering@0.5.0",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "ac64bcfbf3d2749a818eba42bf6bf04d65cd9fea",
    "gitTags": "",
    "committerDate": "2018-09-12 14:55:33 +0100",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.2.2"
  },
  {
    "type": "fix",
    "scope": null,
    "subject": "use @pie-lib/math-rendering instead of @pie-lib/player-utils",
    "merge": null,
    "header": "fix: use @pie-lib/math-rendering instead of @pie-lib/player-utils",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "f82be958df1559e28dfa59964aa20f9aebdbe602",
    "gitTags": "",
    "committerDate": "2018-08-10 12:19:20 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.2.1"
  },
  {
    "type": "feat",
    "scope": null,
    "subject": "add @pie-lib/player-utils#renderMath",
    "merge": null,
    "header": "feat: add @pie-lib/player-utils#renderMath",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "e875aad645b03ca76216ec70fb6cd3d8818157e6",
    "gitTags": " (origin/feature/add-player-utils)",
    "committerDate": "2018-08-08 15:24:08 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.2.0"
  },
  {
    "type": "feat",
    "scope": null,
    "subject": "upgrade material-ui -> @material-ui/core@^1.0.0-rc.1",
    "merge": null,
    "header": "feat: upgrade material-ui -> @material-ui/core@^1.0.0-rc.1",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "ccf025fc93c1abeb9d65c07df0a168341870c8b1",
    "gitTags": "",
    "committerDate": "2018-05-15 22:26:01 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.1.0"
  },
  {
    "type": "chore",
    "scope": "npmignore",
    "subject": "ignore demo",
    "merge": null,
    "header": "chore(npmignore): ignore demo",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "8ba4506439e5e17038c4a9cddc165bc0f342c1d1",
    "gitTags": "",
    "committerDate": "2018-05-02 20:56:27 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.0.1"
  },
  {
    "type": "feat",
    "scope": "build",
    "subject": "This repo now builds the packages for publishing on npm.",
    "merge": null,
    "header": "feat(build): This repo now builds the packages for publishing on npm.",
    "body": null,
    "footer": "BREAKING CHANGE: You no longer build these from src in your destination project. They are pre-built.",
    "notes": [
      {
        "title": "BREAKING CHANGE",
        "text": "You no longer build these from src in your destination project. They are pre-built."
      }
    ],
    "hash": "9a67b2a45d22bde117d2b647aff0d536f27b9356",
    "gitTags": "",
    "committerDate": "2018-05-02 14:38:44 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@4.0.0"
  },
  {
    "type": "fix",
    "scope": "css",
    "subject": "set disabled color",
    "merge": null,
    "header": "fix(css): set disabled color",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "c183cf30a4ae9ba54564b0a52f18ee7f200c3bed",
    "gitTags": "",
    "committerDate": "2018-04-24 12:33:31 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@3.0.2"
  },
  {
    "type": "fix",
    "scope": "lint",
    "subject": "remove duplicate key",
    "merge": null,
    "header": "fix(lint): remove duplicate key",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "5ac6441c864b24a45b3bceac8df09bbd97428c01",
    "gitTags": "",
    "committerDate": "2018-04-24 12:32:46 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@3.0.2"
  },
  {
    "type": "fix",
    "scope": "inputs",
    "subject": "fix styles",
    "merge": null,
    "header": "fix(inputs): fix styles",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "128cc3d45749ab00ba7dbe31f32513af633b8c8e",
    "gitTags": "",
    "committerDate": "2018-04-24 12:32:17 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@3.0.2"
  },
  {
    "type": "fix",
    "scope": "dependencies",
    "subject": "version bump",
    "merge": null,
    "header": "fix(dependencies): version bump",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "0f69ce6e2766fcfe813ade474054f1f2167bb5c8",
    "gitTags": "",
    "committerDate": "2018-04-24 11:24:08 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@3.0.1"
  },
  {
    "type": "fix",
    "scope": "dependencies",
    "subject": "bump dependencies",
    "merge": null,
    "header": "fix(dependencies): bump dependencies",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "a1f4d579f259b09294c48bce217baa06166d6b14",
    "gitTags": "",
    "committerDate": "2018-04-24 09:37:37 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@3.0.1"
  },
  {
    "type": "chore",
    "scope": "test",
    "subject": "tidy up",
    "merge": null,
    "header": "chore(test): tidy up",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "7211be06fc44980e9166e30a74326aca52bee6e4",
    "gitTags": "",
    "committerDate": "2018-04-17 12:38:08 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@3.0.0"
  },
  {
    "type": "fix",
    "scope": "dependencies",
    "subject": "update dependencies",
    "merge": null,
    "header": "fix(dependencies): update dependencies",
    "body": null,
    "footer": "BREAKING CHANGE: use react@16.x",
    "notes": [
      {
        "title": "BREAKING CHANGE",
        "text": "use react@16.x"
      }
    ],
    "hash": "1d8ee3e24b3531faea34f888598a41f246fd6f87",
    "gitTags": "",
    "committerDate": "2018-04-16 15:11:16 -0400",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@3.0.0"
  },
  {
    "type": "fix",
    "scope": "repository",
    "subject": "fix repository",
    "merge": null,
    "header": "fix(repository): fix repository",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "ed63b70a166fd86d0dd8990bf00cd366d12a8bc4",
    "gitTags": "",
    "committerDate": "2018-03-06 14:03:15 -0500",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@2.0.1"
  },
  {
    "type": "fix",
    "scope": "package",
    "subject": "fix repo name",
    "merge": null,
    "header": "fix(package): fix repo name",
    "body": null,
    "footer": null,
    "notes": [],
    "hash": "c96be9698442a375fbc7acbf8ab339501ff961d6",
    "gitTags": "",
    "committerDate": "2018-03-06 13:57:44 -0500",
    "isTagged": true,
    "earliestTag": "@pie-ui/multiple-choice@2.0.1"
  }
]