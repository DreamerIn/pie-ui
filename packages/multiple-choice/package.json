{
  "name": "@pie-ui/multiple-choice",
  "description": "A pie multiple choice component",
  "keywords": [
    "pie",
    "pie-ui",
    "custom-element",
    "react"
  ],
  "version": "4.5.4",
  "main": "lib/index.js",
  "repository": "pie-framework/pie-elements.git",
  "publishConfig": {
    "access": "public"
  },
  "license": "ISC",
  "scripts": {
    "test": "./node_modules/.bin/jest"
  },
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@pie-lib/config-ui": "^7.11.9",
    "@pie-lib/correct-answer-toggle": "^2.1.2",
    "@pie-lib/math-rendering": "^1.2.1",
    "@pie-lib/render-ui": "^4.3.1",
    "classnames": "^2.2.5",
    "debug": "^3.1.0",
    "enzyme-to-json": "^3.3.3",
    "lodash": "^4.17.10",
    "prop-types": "^15.6.1",
    "react": "^16.8.1",
    "react-dom": "^16.8.1",
    "react-test-renderer": "^16.3.2",
    "react-transition-group": "^2.3.1"
  },
  "gitHead": "25b2169d13da6abaaaa962e42f3f6d6ab400ec8b"
}
