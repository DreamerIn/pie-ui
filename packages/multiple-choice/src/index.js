import Main from './main';
import React from 'react';
import ReactDOM from 'react-dom';
import debounce from 'lodash/debounce';
import debug from 'debug';
import { renderMath } from '@pie-lib/math-rendering';
import { updateSessionValue } from './session-updater';

const log = debug('pie-ui:multiple-choice');

const buildAncestors = (n, acc) => {
  if (!acc) {
    acc = [];
  }

  if (n && n.parentNode) {
    acc.push(n.parentNode);
    return buildAncestors(n.parentNode, acc);
  }
  return acc;
};

export default class MultipleChoice extends HTMLElement {
  constructor() {
    super();
    this._model = null;
    this._session = null;

    // this._rerender = debounce(
    //   () => {
    //     console.log('debounced re-render!!!!!!!!!!!!!!!!!!!!!!');
    //     if (this._model && this._session) {
    //       var element = React.createElement(Main, {
    //         model: this._model,
    //         session: this._session,
    //         onChoiceChanged: this._onChange.bind(this)
    //       });
    //       ReactDOM.render(element, this, () => {
    //         log('render complete - render math');
    //         renderMath(this);
    //       });
    //     } else {
    //       log('skip');
    //     }
    //   },
    //   50,
    //   { leading: false, trailing: true }
    // );

    this._rerender = () => {
      console.log('not debounced re-render!!!!!!!!!!!!!!!!!!!!!!');

      const ancestors = buildAncestors(this);

      console.log('document contains this element:', document.contains(this));
      console.log('ancestors:', ancestors);

      if (this._model && this._session) {
        console.log('model/session - render');
        // var element = React.createElement(Main, {
        //   model: this._model,
        //   session: this._session,
        //   onChoiceChanged: this._onChange.bind(this)
        // });
        // ReactDOM.render(element, this, () => {
        //   log('render complete - render math');
        //   renderMath(this);
        // });
        this.innerHTML = JSON.stringify(this._model);
      } else {
        console.log('no model/session - skip');
        log('skip');
      }
    };

    this._dispatchResponseChanged = debounce(() => {
      var event = new CustomEvent('session-changed', {
        bubbles: true,
        composed: true,
        detail: {
          complete: this.isComplete(),
          component: this.tagName.toLowerCase()
        }
      });

      this.dispatchEvent(event);
    });

    this._dispatchModelSet = debounce(
      () => {
        this.dispatchEvent(
          new CustomEvent('model-set', {
            bubbles: true,
            composed: true,
            detail: {
              complete: this.isComplete(),
              component: this.tagName.toLowerCase(),
              hasModel: this._model !== undefined
            }
          })
        );
      },
      50,
      { leading: false, trailing: true }
    );
  }

  set model(s) {
    this._model = s;
    console.log('set model => rerender');
    this._rerender();
    this._dispatchModelSet();
  }

  get session() {
    return this._session;
  }

  set session(s) {
    this._session = s;
    console.log('set session => rerender');
    this._rerender();
    this._dispatchResponseChanged();
  }

  _onChange(data) {
    updateSessionValue(this._session, this._model.choiceMode, data);
    this._dispatchResponseChanged();
    console.log('onChange => rerender');
    this._rerender();
  }

  isComplete() {
    const { complete } = this._model || {};
    if (complete) {
      const { min = -1, max = -1 } = complete;
      const choiceCount =
        this._session && this._session.value ? this._session.value.length : 0;
      const overMin = min === -1 || choiceCount >= min;
      const underMax = max === -1 || choiceCount <= max;
      return overMin && underMax;
    } else {
      return true;
    }
  }

  // connectedCallback() {
  //   console.log('connected => rerender');
  //   this._rerender();
  // }
}
