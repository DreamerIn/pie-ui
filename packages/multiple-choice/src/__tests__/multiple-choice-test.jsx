import ChoiceInput from '../choice-input';
import React from 'react';
import _ from 'lodash';
import { shallow } from 'enzyme';
import { MultipleChoice } from '../multiple-choice';
import toJson from 'enzyme-to-json';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import correctAnswerToggle from '../../__mocks__/@pie-lib/correct-answer-toggle';

describe('CorespringChoice', () => {
  let wrapper, toggle;

  beforeEach(() => {
    toggle = () => {
      return <div>mocked-toggle</div>;
    };
  });

  let mkWrapper = (opts, clone = true) => {
    opts = clone
      ? (opts = _.merge(
          {
            classes: {},
            choices: [],
            disabled: false,
            onChoiceChanged: jest.fn(),
            mode: 'gather'
          },
          opts
        ))
      : opts;

    return shallow(<MultipleChoice {...opts} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      wrapper = mkWrapper({
        choices: [
          { value: 'a', label: 'label a', correct: true, feedback: 'great' },
          { value: 'b', label: 'label b' }
        ]
      });

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('onToggle', () => {
      it('toggles the state', () => {
        let w = mkWrapper({ mode: 'evaluate' });
        expect(w.state('showCorrect')).toEqual(false);
        w.instance().onToggle();
        expect(w.state('showCorrect')).toEqual(true);
      });
    });

    describe('Toggle', () => {
      it('toggle is rendered', () => {
        const t = mkWrapper().find(CorrectAnswerToggle);
        expect(t.length).toEqual(1);
      });

      it('toggle show is set to false', () => {
        expect(
          mkWrapper()
            .find(CorrectAnswerToggle)
            .prop('show')
        ).toEqual(false);
      });

      it('shows toggle if mode is evaluate and responseCorrect is false', () => {
        expect(
          mkWrapper({ mode: 'evaluate', responseCorrect: false })
            .find(CorrectAnswerToggle)
            .prop('show')
        ).toEqual(true);
      });

      it('hides toggle if mode is evaluate and responseCorrect is false', () => {
        expect(
          mkWrapper({ mode: 'evaluate', responseCorrect: true })
            .find(CorrectAnswerToggle)
            .prop('show')
        ).toEqual(false);
      });

      it('not toggled if showCorrect is false', () => {
        let w = mkWrapper({ mode: 'evaluate' });
        w.setState({ showCorrect: false });
        expect(w.find(correctAnswerToggle).prop('toggled')).toEqual(false);
      });

      it('is toggled showCorrect=true && mode=evaluate', () => {
        let w = mkWrapper({ mode: 'evaluate' });
        w.setState({ showCorrect: true });
        expect(w.find(CorrectAnswerToggle).prop('toggled')).toEqual(true);
      });
    });
  });
});
