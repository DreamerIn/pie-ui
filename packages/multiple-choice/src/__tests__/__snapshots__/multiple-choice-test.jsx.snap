// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorespringChoice snapshot renders 1`] = `
<div>
  <Component
    onToggle={[Function]}
    show={false}
    toggled={false}
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <div
    className="choice"
    key="0"
  >
    <WithStyles(ChoiceInput)
      checked={false}
      className=""
      disabled={false}
      displayKey="A"
      feedback=""
      label="label a"
      onChange={[MockFunction]}
      value="a"
    />
  </div>
  <div
    className="choice last"
    key="1"
  >
    <WithStyles(ChoiceInput)
      checked={false}
      className="undefined"
      disabled={false}
      displayKey="B"
      feedback=""
      label="label b"
      onChange={[MockFunction]}
      value="b"
    />
  </div>
</div>
`;
